tests/acceptance/services/before-model-test.js

```diff
@@ -0,0 +1,88 @@
+import Route from '@ember/routing/route';
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { visit } from '@ember/test-helpers';
+import back from '../../helpers/back';
+import replace from '../../helpers/replace';
+
+module('Integration | Service | before-model', (hooks) => {
+	setupApplicationTest(hooks);
+
+	hooks.beforeEach(async function() {
+		await visit('/');
+
+		this.service = this.owner.lookup('service:state');
+	});
+
+	test('it checks replace in beforeModel', async function(assert) {
+		const pointer = this.service.get('current.index');
```

Comment:
This property could be in `beforeEach` hook. It's the same code in every test, doesn´t it?

Reply:
Yep, but I think it is part of the _Arrange_ part of the test itself, so it is good to be here.